File: frontend/src/app/core/breadcrumbs/models/breadcrumb.ts
import { Observable, Subject } from "rxjs";

export type BreadcrumbType = 'navigation' | 'debug' | 'error'; 
export type BreadcrumbCategory = 'react-router' | 'console' | 'ajax' | 'click' | 'exception';
export type BreadcrumbLevel = 'info' | 'warning' | 'error' | 'debug';

export class BreadcrumbBody {

}

export class Breadcrumb {
    type: BreadcrumbType;
    category: BreadcrumbCategory;
    level: BreadcrumbLevel;
    time: Date;
    body: BreadcrumbBody;

    constructor() {
        this.time = new Date;
    }
}

File: backend/Watchdog.Collector/Watchdog.Collector.Common/Models/Breadcrumb.cs
using System;

namespace Watchdog.Collector.Common.Models
{
    public class Breadcrumb
    {
        public string Type { get; set; }
        public string Category { get; set; }
        public string Level { get; set; }
        public DateTime Time { get; set; }
        public string Body { get; set; }
    }
}
